buildshiprun/handler.go

```diff
@@ -146,6 +165,82 @@ func deployFunction(deploy deployment, gatewayURL string, c *http.Client) (strin
 	return string(buildStatus), err
 }
 
+func reportStatus(status string, desc string, statusContext string) {
+
+	if os.Getenv("report_status") != "true" {
+		return
+	}
+
+	url := event.url
+	if status == "success" {
+		publicUrl := os.Getenv("gateway_public_url")
+		// for success status if gateway's public url id set the deployed
+		// function url is used in the commit status
+		if publicUrl != "" {
+			serviceValue := fmt.Sprintf("%s-%s", event.owner, event.service)
+			url = publicUrl + "function/" + serviceValue
+		}
+	}
+
+	repostatus := createStatus(status, desc, statusContext, url)
+
+	ctx := context.Background()
+
+	// NOTE: currently vendored derek auth package doesn't take the private key as input;
```

Comment:
Good note.. let's raise an issue on https://github.com/alexellis/derek for this change and see if @rgee0 can help us